libs/dal/src/repositories/base-repository.ts

```diff
@@ -71,6 +71,22 @@ export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement> {
     return this.mapEntity(data.toObject());
   }
 
+  async findOneAndUpdate(
+    query: FilterQuery<T_DBModel> & T_Enforcement,
+    update: UpdateQuery<T_DBModel>,
+    options: QueryOptions<T_DBModel> = {}
+  ): Promise<T_MappedEntity | null> {
+    const data = await this.MongooseModel.findOneAndUpdate(query, update, {
+      ...options,
+      upsert: options.upsert || false,
+      new: options.new || false,
+    });
+
+    if (!data) return null;
+
+    return this.mapEntity(data.toObject());
+  }
+
```

Comment:
Added the `findOneAndUpdate` method to the `base-repository`
cc: @SokratisVidros